changelog/_unreleased/2025-06-28-fix-credit-note-order-version.md

```diff
@@ -0,0 +1,8 @@
+---
+title: Fix order version for determining credit notes when creating credit notes
+author: Justus Geramb
+author_email: justus@devite.io
+author_github: @jgeramb
+---
+# Core
+* Uses the LIVE version of the order to determine the credit items and then creates a new version to maintain the document state.
```

Comment:
Before merging this, I would wait for a response here:
https://github.com/shopware/shopware/issues/8838#issuecomment-2911376574

cc @ennasus4sun 

Just to be sure there isn't any good reason why it was implemented the other way in the first place